tests/nc-client/main.c

```diff
@@ -38,14 +69,26 @@ send_input(void * cookie)
 			goto err0;
 		}
 	} else {
-		/* Close connection without sending anything. */
-		send->conndone = 1;
-
 		/* If we didn't get an EOF, then there was an error. */
 		if (!feof(stdin)) {
 			warn0("getline(..., ..., stdin)");
 			goto err0;
 		}
+
+		/* We're not going to send anything else. */
+		if (shutdown(send->socket, SHUT_WR)) {
+			warnp("shutdown");
+			goto err0;
+		}
+
+		/* Attempt to read from socket. */
```

Comment:
I'd add a bit more detail here, along the lines of "the server should not send any data back, but attempting to read will detect when the other end of the socket is closed".